modules/core/migrations/v7/genesis.go

```diff
@@ -0,0 +1,40 @@
+package v7
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	clientv7 "github.com/cosmos/ibc-go/v6/modules/core/02-client/migrations/v7"
+	host "github.com/cosmos/ibc-go/v6/modules/core/24-host"
+	"github.com/cosmos/ibc-go/v6/modules/core/types"
+)
+
+// MigrateGenesis accepts an exported IBC client genesis file and migrates it to:
+//
+// - Update solo machine client state protobuf definition (v2 to v3)
+// - Remove all solo machine consensus states
+// - Remove any localhost clients
+func MigrateGenesis(appState genutiltypes.AppMap, cdc codec.ProtoCodecMarshaler) (genutiltypes.AppMap, error) {
+	if appState[host.ModuleName] != nil {
```

Comment:
[nit] I think this could be a little cleaner by reducing the indentation level of the core logic (i.e. early return of `appState` when `appState[host.ModuleName] == nil`)